src/Data/Array/Accelerate/Debug/Internal/Tracy.hs

```diff
@@ -0,0 +1,144 @@
+{-# LANGUAGE CPP                      #-}
+{-# LANGUAGE ForeignFunctionInterface #-}
+{-# LANGUAGE TemplateHaskell          #-}
+{-# OPTIONS_GHC -fobject-code #-}
+-- |
+-- Module      : Data.Array.Accelerate.Debug.Internal.Tracy
+-- Copyright   : [2016..2020] The Accelerate Team
+-- License     : BSD3
+--
+-- Maintainer  : Trevor L. McDonell <trevor.mcdonell@gmail.com>
+-- Stability   : experimental
+-- Portability : non-portable (GHC extensions)
+--
+
+module Data.Array.Accelerate.Debug.Internal.Tracy
+  where
+
+import Data.Word
+import Foreign.C.String
+import Foreign.C.Types
+import Foreign.Ptr
+
+#ifdef ACCELERATE_PROFILE
+import Language.Haskell.TH.Syntax
+#endif
+
+type Zone   = Word64
+type SrcLoc = Word64
+
+#ifdef ACCELERATE_PROFILE
```

Comment:
For these and the other usages of the FFI, we might also need to do that `__GHCIDE__` check to make sure that hls/ghcide doesn't crash on us when another module includes this module. So then this would need to become `#if defined(ACCELERATE_PROFILE) && !defined(__GHCIDE__)`. Or did that get silently fixed in hls 1.2.0?

Reply:
I don't hls/ghcide so didn't try, but happy to add it (and hope for the best)